src/Language/PureScript/Errors.hs

```diff
@@ -1306,6 +1316,25 @@ prettyPrintSingleError (PPEOptions codeColor full level showDocs relPath) e = fl
         , line "Instance declarations of this type class are disallowed."
         ]
 
+    renderSimpleErrorMessage UnsupportedRoleDeclaration =
+      line $ "Role declarations are only supported for data types, not for type synonyms nor type classes."
+
+    renderSimpleErrorMessage (RoleDeclarationArityMismatch name expected actual) =
+      line $ T.intercalate " "
+        [ "The type"
+        , markCode (runProperName name)
+        , "expects"
+        , T.pack (show expected)
+        , if expected > 1 then "arguments" else "argument"
+        , "but its role declaration list"
```

Comment:
To make the verb "list" agree, it should be written "lists", so for example: "but its role declarations lists only 3 roles".